compat-data-schema.md

```diff
@@ -0,0 +1,365 @@
+
+__Note:__ The schema tries to capture as many constraints and rules about the
+browser-compat-data JSON format. Nevertheless there are constraints that cannot
+be described inside a schema, and won't be checked by automated tests. We tried
+to mark such cases in the following text with a __Not enforced by the
+schema__ notice.
+
+## The JSON format
+
+### Division in files
+The browser-compat-data schema has been designed so the division in files doesn't
+convey any meaning. That means that browser compatibility information about features
+can be stored in one single large files or being divided in smaller files.
+
+The division in separate files, themselves in different directories is guided
+by making it easy to maintain by humans.
+
+### Schema versioning
+There one single information that is tied to each file, and is mandatory. It is
+the version of schema used inside the file.
+
+    {
+      "version": "1.0.0",
+      "data": {
+        …
+      }
+    }
+
+We are using semantic versioning, so the version is a `string` containing three
+integers separate by a dot (`.`). No whitespace or additional characters are
+allowed. To see detail of the meaing fo the 3 integers, see the
+[Semantic Versioning 2.0.0 specification](http://semver.org/).
+
+The "data" contains a list of _features_, with their identifier, their description,
+their support status, and thir status.
+
+__Note:__ You cannot mix two schemas in the same file. `"version"` is unique in
+a given file.
+
+### Feature identifier
+A _feature_ is a functionality of the platform. It is an entity that can be used
+independently. A _sub-feature_ is a specific value or behavior of a feature. The
+division is a bit arbitrary, but matches the way developers perceive features of
+a platform
+
+* For CSS, an entity like a property, a pseudo-class, a pseudo-element, an
+at-rule, or a descriptor is a feature. A value, a new syntax change, or a
+specific behavior, like if a property applies to a set of elements or another
+are sub-features.
+
+* For HTML, an element or an attribute are features, while specific values of an
+attribute are sub-features.
+
+* For Web APIs or JavaScript, an interface, a method, a property, a constructor
+are features, while arguments or special values of an enumerated type are
+sub-features.
+
+Each feature is identified by a unique hierarchy of strings. E.g the
+`text-align` property is identified by `css.properties.text-align`.
+
+The strings of an identifier are not intended to be displayed and are therefore
+not translatable.
+
+In JSON, this gives:
+
+    "data": {
+      "css": {
+        "properties": {
+          "text-align": {…},
+          …
+        },
+        …
+      },
+      …
+    }
+
+ That way, each feature is uniquely identifier, independently of the file it is
+ defined in.
+
+ The hierarchy of identifiers is not defined inside the schema. It is a
+ convention of the project using the schema.
+
+ In the MDN browser-compat case, it is:
```

Comment:
I'm not sure we should say this. It's already out of date :). We could just say "for example" instead, or omit it.